### PR TITLE
Task 46447: disable edit inline on Task fields.

### PR DESCRIPTION
**Pull request by wassimmhamdi**
before this fix, the auto save function of task description work with no limitation of number of characters , in this fix I add a button to assure that the field of description must be under than a maximum number of a characters given which is 2000 and when we write more than 2000 character the button will be disabled and we can't save our changes

### CODE REVIEW

**Review by rdenarie — CHANGES_REQUESTED** [review]
Can you fix the conflict ? 
Thanks

File: task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawerComponents/TaskDescriptionEditor.vue [hunk 1]
```diff
@@ -129,7 +132,15 @@ export default {
     });
   },
   methods: {
+    disableApply (val){
```
Comment (rdenarie) [inline task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawerComponents/TaskDescriptionEditor.vue]:
```suggestion
    updateButtonEnablement (val){
```

File: task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawerComponents/TaskDescriptionEditor.vue [hunk 2]
```diff
@@ -19,24 +19,26 @@
...
-      {{ placeholder }}
-    </div>
-
     <exo-task-editor
       v-if="displayEditor"
       ref="richEditor"
       v-model="inputVal"
       :max-length="MESSAGE_MAX_LENGTH"
       :id="task.id"
+      @counterChanged="disableApply"
```
Comment (rdenarie) [inline task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawerComponents/TaskDescriptionEditor.vue]:
```suggestion
      @counterChanged="updateButtonEnablement"
```

File: task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawerComponents/TaskDescriptionEditor.vue [hunk 3]
```diff
@@ -19,24 +19,26 @@
...
       :placeholder="$t('task.placeholder').replace('{0}', MESSAGE_MAX_LENGTH)" />
 
+    <v-btn
+      v-if="task.id!=null"
+      id="apply-btn"
```
Comment (rdenarie) [inline task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawerComponents/TaskDescriptionEditor.vue]:
```suggestion
      id="saveDescriptionButton"
```

File: task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawerComponents/TaskDescriptionEditor.vue [hunk 4]
```diff
@@ -19,24 +19,26 @@
...
+      depressed
+      outlined
+      class="btn mt-1 ml-auto d-flex"
```
Comment (rdenarie) [inline task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawerComponents/TaskDescriptionEditor.vue]: Should we have primary color on this button ?

File: task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawerComponents/TaskDescriptionEditor.vue [hunk 5]
```diff
@@ -19,24 +19,26 @@
...
+      :disabled="buttonOff"
+      @click="saveDescription(inputVal)"
+    >
+      Apply
```
Comment (rdenarie) [inline task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawerComponents/TaskDescriptionEditor.vue]: This line must be internationnalized. In addition, the string should be something like "Save"

File: task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawerComponents/TaskDescriptionEditor.vue [hunk 6]
```diff
@@ -129,7 +132,15 @@ export default {
     });
   },
   methods: {
+    disableApply (val){
+      this.buttonOff = !(val<= this.MESSAGE_MAX_LENGTH);
+    },
     saveDescription: function (newValue) {
+      let pureText = newValue ? newValue.replace(/<[^>]*>/g, '').replace(/&nbsp;/g, '').trim() : '';
```
Comment (rdenarie) [inline task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawerComponents/TaskDescriptionEditor.vue]: This completly remove the text formatting
For example, if I add new lines in the description, and save, new lines are lost.
We need to keep the html markup and count it in the character counter

**Review by rdenarie — DISMISSED** [review]
I add comments inline. 
There are 2 others points to address :

- When I open the task the counter is set to 0 even if there is some chars in description
- When I open the task, the field is alread in edit mode. But I dont want to modify it. So we need to keep the description displayed when the task is opened, and when I click into the description, we go in edit mode and display the save button. When I click on save button, we should remove the edit mode and go back in display mode

File: task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawer.vue [hunk 7]
```diff
@@ -256,6 +254,10 @@ export default {
       oldTask: {},
       showBackArrow: false,
       taskSpace: {},
+      taskTitle: null,
+      taskDescription: null,
+      descriptionValid: false,
+      buttonOff: false,
```
Comment (boubaker) [inline task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawer.vue]:
```suggestion
```
not used variables

File: task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawer.vue [hunk 8]
```diff
@@ -347,6 +350,9 @@ export default {
     document.removeEventListener('keyup', this.escapeKeyListener);
   },
   methods: {
+    saveButtonEnablement(val) {
+      this.buttonOff = val > this.MESSAGE_MAX_LENGTH;
+    },
```
Comment (boubaker) [inline task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawer.vue]: not used method

File: task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawerComponents/TaskDescriptionEditor.vue [hunk 9]
```diff
@@ -179,20 +195,23 @@ export default {
...
           },
           destroy: function () {
-            self.inputVal = '';
             self.editorReady = false;
+            self.inputVal = '';
           }
         },
       });
     },
+    hideDescriptionEditor() {
```
Comment (boubaker) [inline task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawerComponents/TaskDescriptionEditor.vue:216]: useful method but not used in current component, can you please use it to hide Editor

File: task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawerComponents/TaskDescriptionEditor.vue [hunk 10]
```diff
@@ -28,24 +28,39 @@
...
+        :class="charsCount > MESSAGE_MAX_LENGTH ? 'tooManyChars' : ''"
+        class="activityCharsCount flex d-flex justify-end"
+        style="">
+        {{ charsCount }}/{{MESSAGE_MAX_LENGTH}}
+      <i class="uiIconMessageLength"></i>
+      </div>
+
+
+
+
```
Comment (boubaker) [inline task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawerComponents/TaskDescriptionEditor.vue]: please execute `npm run lint` to format files

File: task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawer.vue (hunk 7 above)
Comment (wassimmhamdi) [reply task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawer.vue]: resolved

File: task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawer.vue (hunk 8 above)
Comment (wassimmhamdi) [reply task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawer.vue]: resolved

File: task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawerComponents/TaskDescriptionEditor.vue (hunk 9 above)
Comment (wassimmhamdi) [reply task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawerComponents/TaskDescriptionEditor.vue:216]: I did it

File: task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawerComponents/TaskDescriptionEditor.vue (hunk 10 above)
Comment (wassimmhamdi) [reply task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawerComponents/TaskDescriptionEditor.vue]: I did It

**Review by boubaker — APPROVED** [review]

**Pull request merged**